akka-http-core/src/main/scala/akka/http/impl/model/JavaUri.scala

```diff
@@ -21,7 +21,7 @@ case class JavaUri(uri: sm.Uri) extends jm.Uri {
 
   def scheme(): String = uri.scheme
   def host(): jm.Host = uri.authority.host
-  def port(): Int = uri.authority.port
+  def port(): Int = uri.effectivePort
```

Comment:
My thinking was that: in the Scala API we cannot hide the impl-detail-port because of binary compatibility, but in this API we are free to hide it.

Can you see a use case when it would be useful to access it?